$MirOS: ports/mailnews/pine/patches/patch-pico_pico_c,v 1.1.7.1 2005/03/18 15:49:48 tg Exp $
$OpenBSD: patch-pico_pico_c,v 1.2 2003/10/05 21:42:29 jakob Exp $
--- pico/pico.c.orig	Thu Mar 31 17:08:35 2005
+++ pico/pico.c	Wed May  4 23:28:08 2005
@@ -876,8 +876,11 @@ stripwhitespace()
     LINE *cur_line = lforw(curbp->b_linep);
 
     do{
+      /* test for lines starting with space or dash (patches) */
+      if(!(cur_line->l_used && (isspace(lgetc(cur_line, 0).c)
+	   || lgetc(cur_line, 0).c == '-'))
       /* we gotta test for the sigdash case here */
-      if(!(cur_line->l_used == 3 && 
+	   || !(cur_line->l_used == 3 && 
 	   lgetc(cur_line, 0).c == '-' &&
 	   lgetc(cur_line, 1).c == '-' &&
 	   lgetc(cur_line, 2).c == ' '))
